test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -138,35 +144,27 @@ public void testStreamContents(InputStreamProvider inputStreamProvider,
      */
     @Test
     public void testMoreThanTwoGB() throws IOException {
-        Path sourceFile = Files.createTempFile("test2GBSource", null);
-        try {
-            // preparing two temporary files which will be compared at the end of the test
-            Path targetFile = Files.createTempFile("test2GBtarget", null);
-            try {
-                // writing 3 GB of random bytes into source file
-                for (int i = 0; i < NUM_WRITES; i++)
-                    Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
-
-                // performing actual transfer, effectively by multiple invocations of Filechannel.transferTo(FileChannel)
-                long count;
-                try (InputStream inputStream = Channels.newInputStream(FileChannel.open(sourceFile));
-                     OutputStream outputStream = Channels
-                             .newOutputStream(FileChannel.open(targetFile, StandardOpenOption.WRITE))) {
-                    count = inputStream.transferTo(outputStream);
-                }
-
-                // comparing reported transferred bytes, must be 3 GB
-                assertEquals(count, BYTES_WRITTEN);
-
-                // comparing content of both files, failing in case of any difference
-                assertEquals(Files.mismatch(sourceFile, targetFile), -1);
-
-            } finally {
-                 Files.delete(targetFile);
-            }
-        } finally {
-            Files.delete(sourceFile);
+        // preparing two temporary files which will be compared at the end of the test
```

Comment:
I would prefer to keep the previous changes to this test.  We went through a few iterations off line.  We would also prefer to delete these files at the end of the test, not just the over all test run due to their size.

Reply:
As all the *other* tests in this test class run rather quick (within seconds) I wonder what you expect to *actually* gain? The difference would just be some seconds of earlier deletion for the sake of worse readability, actually. If the majority is for keeping your original solution, this is fine for me. I just like to understand the driver behind this.

Reply:
In the unlikely event there is an error, we can uniquely identify which file is left around more easily.  Your proposed change ignores the IOException completely.

Given the size of the files, best to clean up before existing the test method.

Reply:
What should it be good for knowing which test left behind the temp file? As the code is not incorrect, you always would just delete *all* temp files of this test anyways - whether they are from the 2GB test or not. So this information helps you exactly nothing.

Your original code also ignored the IOException completely, so I do not understand that argument.

I do not see *any* benefit from immediate deletion vs deletion after ten seconds.

Those arguments are rather academic. I miss *real* benefit, actually.